tests/test_main.py

```diff
@@ -0,0 +1,59 @@
+from pathlib import Path
+
+import inspect
+import ast
+
+from app.main import calculate_profit
+
+from app import main
+
+BASE_DIR = Path(__file__).resolve().parent.parent
+
+
+def test_should_be_declared():
+    assert (
+        hasattr(main, "calculate_profit") is True
+    ), "Function 'calculate_profit' should be declared."
```

Comment:
Unnecessary test.
You have import `calculate_profit` from app.main higher. So if it is not declared - the error will be on line 6